docs/source/config/inputtransforms.rst

```diff
@@ -71,6 +71,32 @@ The decorator returns a factory function which will produce instances of
 :class:`~IPython.core.inputtransformer.StatelessInputTransformer` using your
 function.
 
+Transforming a full block
+-------------------------
+
+Transforming a full block of python code is possible by implementing a
+:class:`~IPython.core.inputtransformer.Inputtransformer` and overwriting the
+``push`` and ``reset`` methods. The reset method should send the full block of
+transformed text. As an example a transformer the reversed the lines from last
+to first.
```

Comment:
We should say that this will break the automatic 'is this input complete' detection in console interfaces - with a transformer like this added, it will always think more input is needed, and you will need to use a shortcut to force it to execute.